presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java

```diff
@@ -19,8 +19,30 @@
 
 public interface ConnectorSplitManager
 {
+    /**
+     * @deprecated use {@link #getSplits(ConnectorTransactionHandle, ConnectorSession, ConnectorTableLayoutHandle, SplitSchedulingStrategy)} instead.
+     */
+    @Deprecated
     default ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout)
     {
         throw new UnsupportedOperationException("not yet implemented");
     }
+
+    default ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout, SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        if (splitSchedulingStrategy == SplitSchedulingStrategy.ALL_AT_ONCE) {
+            return getSplits(transactionHandle, session, layout);
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    default boolean supportsGroupedScheduling(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layout)
+    {
+        return false;
+    }
+
+    enum SplitSchedulingStrategy
+    {
+        ALL_AT_ONCE,
```

Comment:
Instead of ALL_AT_ONCE, SINGLE_GROUP? "All at once" seems to imply they will all get scheduled at the same time.